packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js

```diff
@@ -34,12 +37,32 @@ export default async (req, res) => {
         break
       case "payment_intent.amount_capturable_updated":
         if (!order) {
-          await manager.transaction(async (manager) => {
-            const cartServiceTx = cartService.withTransaction(manager)
-            await cartServiceTx.setPaymentSession(cartId, "stripe")
-            await cartServiceTx.authorizePayment(cartId)
-            await orderService.withTransaction(manager).createFromCart(cartId)
-          })
+          const err = await manager
+            .transaction(async (manager) => {
+              const cartServiceTx = cartService.withTransaction(manager)
+              await cartServiceTx.setPaymentSession(cartId, "stripe")
+              await cartServiceTx.authorizePayment(cartId)
+              await orderService.withTransaction(manager).createFromCart(cartId)
+            })
+            .catch((e) => {
+              return e
+            })
+
+          if (err) {
+            let message = `Stripe webhook ${event.type} handling failed\n${
+              err?.detail ?? err?.message
+            }`
+            if (err.code === PostgresError.SERIALIZATION_FAILURE) {
+              message = `Stripe webhook ${
+                event.type
+              } handle failed. This can happen when this webhook is triggered during a cart completion and can be ignored. This event should be retried automatically.\n${
+                err?.detail ?? err?.message
+              }`
+            }
+            logger.warning(message)
```

Comment:
thought(non-blocking): this seems like a lot of work to put on the shoulders of the plugin author; would love for us to figure out a better way to handle payments so it becomes easier to build plugins that are reliable and robust. 

--
There is no action expected from my side - just an observation :)

Reply:
It seams that with the stripe webhook and serializable transaction we don’t have much choice for now. When the payment will be re worked and maybe they will be async, we will probably have the same kind of behaviour to manage as we will probably want the payment provider work to be serialisable and retriable. Here the intent is to let stripe try again later. Also, i think that as a plugin author working with payment provider we can’t avoid that much to expect them to manage race condition such as a webhook triggered while doing some other actions, do you have something in mind?

Reply:
I would rather see that we rever to this version and rethink this flow separately than the proposal you have in the latest commits;

Reply:
Alright, i ll revert it

Reply:
done